src/pages/atomicdex/api/v20-dev/non_fungible_tokens/index.mdx

```diff
@@ -0,0 +1,577 @@
+| withdraw_type | object           | A standard [WithdrawType](/atomicdex/api/v20#withdraw-type) object.                                       |
+
+
+#### 📌 ERC721 Withdraw Example
+
+<CodeGroup title="" tag="POST" label="withdraw_nft">
+```json {{ mm2MethodDecorate : true }}
+{
+  "userpass": "MM2_RPC_PASSWORD",
+  "method": "withdraw_nft",
+  "mmrpc": "2.0",
+  "params": {
+    "url": "https://moralis-proxy.komodo.earth",
+    "withdraw_type": {
+      "type": "withdraw_erc721",
+      "withdraw_data": {
+        "chain": "BSC",
+        "to": "0x6FAD0eC6bb76914b2a2a800686acc22970645820",
+        "token_address": "0xfd913a305d70a60aac4faac70c739563738e1f81",
+        "token_id": "214300044414"
+      }
+    }
+  }
+}
+```
+</CodeGroup>
+
+### Response Parameters
+
+| Parameter            | Type             | Description                                                                                      |
+| ---------------------| ---------------- | ------------------------------------------------------------------------------------------------ |
+| amount               | string           | Amount of tokens to withdraw.                                                                    |
+| tx_hex               | string           | Raw transaction hex of the withdrawl.                                                            |
+| tx_hash              | string           | Transaction ID of the withdrawl.                                                                 |
+| from                 | array            | List of source addresses.                                                                        |
+| to                   | array            | List of destination addresses.                                                                   |
+| contract_type        | string           | Contract type. `ERC721` or `ERC1155`.                                                            |
+| token_address        | string           | Token address.                                                                                   |
+| token_id             | string           | Token ID.                                                                                        |
+| fee_details          | object           | A standard [WithdrawFee](/atomicdex/api/v20#withdraw-fee) object.                                |
+| coin                 | string           | Coin name. One of `AVALANCHE`, `BSC`, `ETH`, `FANTOM`, or `POLYGON`. CONFIRM: EVERYWHERE ELSE THIS PARAM IS `chain`.                            |
+| block_height         | integer          | Block height of the transaction. If the value is `0`, the transaction is unconfirmed.            |
+| timestamp            | integer          | Timestamp of the block containing the withdrawl transaction.                                     |
+| internal_id          | integer          | I HAVE NO IDEA WHAT THIS MEANS.                                                                  |
```

Comment:
yeah, all our structures for Transaction details have this field.
`Internal MM2 id used for internal transaction identification, for some coins it might be equal to transaction hash`

This is how it looks in [TransactionNftDetails](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/nft/nft_structs.rs#L214)

When user generates nft transaction [I set 0](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L980)

It is also empty when [user generates tx to withdraw ETH coins, this is for example, other coins also use this structure](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L868). Actually this is the response for [withdraw method](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L868)

But we fill it during [processing erc20 history](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L2940)
here the [process_erc20_history function description ](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/coins/eth.rs#L2608)

So the response for our general withdraw method also contains this field
https://docs.komodoplatform.com/basic-docs/atomicdex-api-20/withdraw.html but its just not documented.

And yeah, we fill it for our internal identification of transaction, zcoin also has it in Tx details and we use it for [history](https://docs.komodoplatform.com/basic-docs/atomicdex-api-20-dev/zhtlc_coins.html#z-coin-tx-history)

Right now it is always zero and actually I dont save TransactionNftDetails in db as we use info from moralis (NftTransferHistory), but may be in the future. I will need it...

Reply:
Updated description

Reply:
added devnote in case of future changes